OttoTest/app/src/main/java/com/example/raulstriglio/ottotest/OttoMainActivity.java

```diff
@@ -1,35 +1,93 @@
 package com.example.raulstriglio.ottotest;
 
+import android.app.FragmentManager;
 import android.app.FragmentTransaction;
-import android.content.Intent;
+import android.app.Fragment;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 
+import com.example.raulstriglio.ottotest.Fragments.DetailFragment;
+import com.example.raulstriglio.ottotest.Fragments.OttoFragment;
+import com.example.raulstriglio.ottotest.Fragments.OttoFragmentList;
+
 import com.example.raulstriglio.ottotest.model.User;
 
 import java.util.ArrayList;
-import java.util.List;
 
-public class OttoMainActivity extends AppCompatActivity implements OttoFragment.ICallback {
+public class OttoMainActivity extends AppCompatActivity implements OttoFragment.ICallback<User>, OttoFragmentList.CallbackToFragmentList {
+
+    public static final String EXTRA_KEY = "userList";
+
+    public static final String NAME_KEY = "name";
+    public static final String LAST_NAME_KEY = "lastname";
 
     OttoFragment ottoFragment = new OttoFragment();
+    ArrayList<User> userList;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_otto_main);
 
         FragmentTransaction ft = getFragmentManager().beginTransaction();
-        ft.add(R.id.FrameID, ottoFragment);
-        ft.addToBackStack(null);
+        ft.add(R.id.FrameID, ottoFragment, ottoFragment.TAG);
+        ft.addToBackStack(ottoFragment.TAG);
         ft.commit();
 
     }
 
     @Override
     public void initListActivity(ArrayList<User> userList) {
-        Intent intent = new Intent(getApplicationContext(), ListActivity.class);
-        intent.putExtra(ListActivity.EXTRA_KEY, userList);
-        startActivity(intent);
+
+        this.userList = userList;
+        OttoFragmentList ottoListFragment = new OttoFragmentList();
```

Comment:
Please use static initialization
http://stackoverflow.com/questions/9245408/best-practice-for-instantiating-a-new-android-fragment